app/gui/src/model/execution_context/synchronized.rs

```diff
@@ -95,6 +99,15 @@ impl ExecutionContext {
         result.map(|res| res.map_err(|err| err.into()))
     }
 
+    /// Load the component groups defined in libraries imported into the execution context.
+    async fn load_component_groups(&self) -> FallibleResult {
+        let ls_response = self.language_server.get_component_groups(&self.id).await?;
+        let ls_component_groups = ls_response.component_groups.into_iter();
+        let component_groups = ls_component_groups.map(|group| group.into());
+        *self.model.component_groups.borrow_mut() = component_groups.collect();
```

Comment:
This code will be clearer this way:

```
*self.model.component_groups.borrow_mut() =
    ls_response.component_groups
        .into_iter()
        .map(|group| group.into())
        .collect()
```
I know this is against our rule of one variable per line, but in this case, these transformations are so obvious that keeping only transformations is easier to read than big var names. I am not yet sure how we should change our Rust style guide for that, but there are cases where sequence of operations (like in my snippet above) is better than our current guideline.